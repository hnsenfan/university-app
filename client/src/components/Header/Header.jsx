import { NavLink, withRouter } from 'react-router-dom'
import PropTypes from 'prop-types'

import './Header.scss'

const Header = ({ location }) => {
  return (
    <header className='bg-white shadow pv-l'>
      <div className='d-flex align-items-center ml-xl'>
        <div className='d-flex align-items-center flex-basis-50'>
          University App
        </div>
        { location.pathname !== '/' &&
          <div className='d-flex align-items-center justify-content-between flex-basis-50 header'>
            <NavLink exact activeClassName="active" to="/home" className='item'>
              <label className='text-bold text-18 pointer'>Dashboard</label>
            </NavLink>
            <NavLink exact activeClassName="active" to="/universities" className='item'>
              <label className='text-bold text-18 pointer'>Universities</label>
            </NavLink>
            <NavLink exact activeClassName="active" to="/more" className='item'>
              <label className='text-bold text-18 pointer'>Profile</label>
            </NavLink>
          </div>
        }
      </div>
    </header>
  )
}

Header.propTypes = {
  history: PropTypes.object,
  location: PropTypes.object,
  hasBack: PropTypes.bool,
  title: PropTypes.string
}

export default withRouter(Header)