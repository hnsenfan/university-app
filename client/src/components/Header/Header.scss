@import '/src/modules/styles/core/variables';

.header-bar {
  height: 48px;
}

.header {
  @media (max-width: $screen-md) {
    display: none !important;
  }
}
  