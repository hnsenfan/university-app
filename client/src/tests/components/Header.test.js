import { shallow } from 'enzyme'
import Header from '../../components/Header/Header'

test('should test Header component', () => {
 const wrapper = shallow(<Header.WrappedComponent />)
 expect(wrapper).toMatchSnapshot()
})