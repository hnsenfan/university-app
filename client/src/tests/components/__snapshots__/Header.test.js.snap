// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test Header component 1`] = `
<header
  className="bg-white shadow pv-s header-bar"
>
  <div
    className="d-flex align-items-center ml-xl"
  >
    <div
      className="d-flex align-items-center flex-basis-50 pv-m"
    >
      <NavLink
        to="/home"
      >
        <label
          className="text-italic text-semibold text-black pointer ada"
        >
          University App
        </label>
      </NavLink>
    </div>
    <div
      className="d-flex align-items-center justify-content-between flex-basis-50 header"
    >
      <NavLink
        activeClassName="active"
        className="item"
        exact={true}
        to="/home"
      >
        <label
          className="text-bold text-18 pointer"
        >
          Dashboard
        </label>
      </NavLink>
      <NavLink
        activeClassName="active"
        className="item"
        exact={true}
        to="/universities"
      >
        <label
          className="text-bold text-18 pointer"
        >
          Universities
        </label>
      </NavLink>
      <NavLink
        activeClassName="active"
        className="item"
        exact={true}
        to="/profile"
      >
        <label
          className="text-bold text-18 pointer"
        >
          Profile
        </label>
      </NavLink>
    </div>
  </div>
</header>
`;
